### PR TITLE
#170947679 Send notification to the manager when request is edited

### PR DESCRIPTION
**Pull request by EugeneMunya**
#### What does this PR do?
- Send notification to the user and his/her manager when a request is modified
#### Description of Task to be completed?
- When a requester modifies something on the trip request he/she has made, notification will be
pop up and notification panel will be updated to both requester and his/ her manager, the notification will be clickable and redirect the user to the edited trip request.

#### How should this be manually tested?
- open your terminal and run this command `git clone https://github.com/andela/knightsbn    backend.git` to clone repository
- run `cd knights-bn-backend` to get inside the project
- run `git fetch && git checkout ft-receive-notification-when-edit-170947679` 
- run `npm install` to install dependencies
- check if you have PostgreSQL database if you don't have it install it from here [PostgreSQL](https://www.2ndquadrant.com/en/blog/pginstaller-install-postgresql/)
- create database
- create `.env` file in your project root directory
- follow `env.example` file to create environment variables in `.env` file and 
   assign them with all your database configurations 
- run `npm run createTables` to create tables in the new database you have created.
- run `npm run dev-start` to start the server.
- open your postman or any other API tester you have and use this URL to login `http://localhost:4000/api/v1/auth/login` and down below there is a default user you can use
```
{
   "email": "alain.maxime@gmail.com",
    "password": "Niyonkuru@1"
}
```
- open this URL in your browser `http://localhost:4000/client/editRequest.html`
allow notification permission first by clicking on allow button on the browser alert which looks like the image below, you can also use this page to view all the trip request you have so that you can edit them.
Screen Shot 2020-03-17 at 07 56 30

- use postman and send PATCH request to this URL`http://localhost:4000/api/v1/trips/edit/{id}` to update trip request of your choice.

#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
[#170947679](https://www.pivotaltracker.com/story/show/170947679)
#### Notification
Screen Shot 2020-03-17 at 13 03 50

#### Update notification panel
Screen Shot 2020-03-17 at 13 03 36

#### Redirect to the trip request
Screen Shot 2020-03-13 at 09 26 12

### CODE REVIEW

**Comment by Niyonsengaeric** [comment]
@EugeneMunya  Look at the error it is showing to me
Screen Shot 2020-03-17 at 10 01 50

**Comment by T2Wil** [comment]
@EugeneMunya , The link provided is not redirecting to the request but to edit request.
You may fix this.

image

**Comment by T2Wil** [comment]
@EugeneMunya , why should I fill in  all fields to make an edit 

image

**Comment by EugeneMunya** [comment]
> @EugeneMunya Look at the error it is showing to me
> Screen Shot 2020-03-17 at 10 01 50

@Erik can you please download the original PR on my end I dont see those errors

**Comment by EugeneMunya** [comment]
> @EugeneMunya , why should I fill in all fields to make an edit
> 
> image

That is the simulation on how  the front end will work,  you can also use Postman or Swagger but make sure to open that in order to receive notification

**Comment by EugeneMunya** [comment]
> @EugeneMunya , The link provided is not redirecting to the request but to edit request.
> You may fix this.
> 
> image

@T2Wil let me resolve it

**Comment by EugeneMunya** [comment]
> @EugeneMunya , The link provided is not redirecting to the request but to edit request.
> You may fix this.
> 
> image

@T2Wil  that what I had to do,  just to redirect user to edited request, not the previous one

**Review by T2Wil — APPROVED** [review]

**Review by Kagorora — APPROVED** [review]

File: .env.example [hunk 1]
```diff
@@ -15,4 +15,6 @@ FCBK_CLIENT_SECRET = 'your-fb-secret'
 FCBK_CALLBACK_URL = 'https://your-heroku-app-on-facebook'
 HOST_NAME='your-heroku-staging-app'
 LOCAL_HOST='http://localhost:port'
-
+CLOUD_NAME=dktzolahb
+API_KEY=292262139124291
+API_SECRET=2eNUENGoMV1zw95hqfOjsNAeHwY
```
Comment (kagabof) [inline .env.example]: Please add a new line at the end of this file.

File: public/client/editRequest.html [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+      <ul id="notification"></ul>
+      <div id="response">
+      </div>
+        <script src="https://cdnjs.cloudflare.com/ajax/libs/socket.io/2.0.0/socket.io.js"></script>
+        <script src="./notification.js"></script>
+        <script src="./getAllRequest.js"></script>
+      <script src="./main.js"></script>
+      <script src="./editEvent.js"></script>
+    </body>
+</html>
```
Comment (kagabof) [inline public/client/editRequest.html]: Please add a new line at the end of this file.

**Review by kagabof — CHANGES_REQUESTED** [review]

File: public/client/getAllRequest.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+const result = document.getElementById('response');
```
Comment (kagabof) [inline public/client/getAllRequest.js]: Please explain what you are trying to do in this file.

**Review by kagabof — CHANGES_REQUESTED** [review]

File: public/client/getAllRequest.js (hunk 3 above)
Comment (EugeneMunya) [reply public/client/getAllRequest.js]: I wanted some who is testing this PR not to wast his time for searching an ID of his trip request he wants to modify  if can you see when you are logged in you can see all your trip requests and you can get ID there and update it very easily

Comment (EugeneMunya) [reply public/client/getAllRequest.js]: I have removed it it does not play an important role here

**Review by kagabof — APPROVED** [review]

**Pull request merged**